### PR TITLE
Add a test for invalid c arguments with bindgen

### PR DESCRIPTION
**Pull request by dcbaker**

### CODE REVIEW

File: test cases/rust/12 bindgen/meson.build [hunk 1]
```diff
@@ -33,69 +37,76 @@ gen = rust.bindgen(
...
-gp_lib = static_library('gp_lib', 'src/global.c')
-
-gp_exe = executable(
-  'gp_exe',
-  structured_sources(['src/global.rs', gp]),
-  link_with : gp_lib,
-)
-
-test('global and project arguments', gp_exe)
+if not get_option('bad-values')
```
Comment (nirbheek) [inline test cases/rust/12 bindgen/meson.build:40]: Why is this conditional needed? I don't understand it. Shouldn't all the code be resistant to CFLAGS that it doesn't understand as long as those are valid CFLAGS?

Comment (dcbaker) [reply test cases/rust/12 bindgen/meson.build:40]: The problem is that it tries to initiate the c compiler with the bad flag, and then fails. Maybe this level of complexity isn’t worth it compared to just writing a plain unit test

Comment (nirbheek) [reply test cases/rust/12 bindgen/meson.build:40]: The problem seems to be compiler flags that aren't understood by the Clang compiler, fwict. So it should be possible to test it like so:

```
cc_id = cc.get_id()
if cc_id == 'gcc'
  bad_values = ['-mtls-dialect=gnu2']
elif cc_id == 'msvc'
  bad_values = ['/fp:fast']
endif
```

Flags like `-fmath-errno` that are understood by both GCC and Clang do not cause any issues.

Comment (nirbheek) [reply test cases/rust/12 bindgen/meson.build]: I pushed a fixup commit that implements this. I tested that the tests fail if I revert the fixes.

**Pull request merged**